### ISSUE TITLE
Not able to ping some docker containers from host

### ISSUE THREAD

**Issue by rohit4813** [moby/moby#20237]
We are trying to setup over 1000 docker containers on one of our servers. However, when we try to ping the IP address for some of the containers from the host, we get "Destination Host unreachable". There are other containers which can be pinged from the host.

Upon further debugging we see that ARP request for these unreachable containers are failing.

Running Ubuntu 15.10 and docker 1.10
![screenshot_from_2016-02-11_23_38_55](https://cloud.githubusercontent.com/assets/13277338/13003890/0ace201e-d19e-11e5-8551-099306625089.png)

![screenshot_from_2016-02-11_22_00_14 1](https://cloud.githubusercontent.com/assets/13277338/13003893/120f1e5a-d19e-11e5-8ff1-6857f0a81737.png)

docker version: 
Client:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   590d5108
 Built:        Thu Feb  4 18:41:30 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   590d5108
 Built:        Thu Feb  4 18:41:30 2016
 OS/Arch:      linux/amd64

docker info:
Containers: 1023
 Running: 1023
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.10.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 2060
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.2.8-ckt1
Operating System: Ubuntu 15.10
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 62.74 GiB
Name: st1
ID: U3E4:LSBV:JSYS:2KKL:RB3U:RMJC:OCJW:VHX3:PVKB:MWIY:WXC6:DCMW

uname -a:
Linux st1 4.2.8-ckt1 #1 SMP Tue Feb 9 17:31:37 CET 2016 x86_64 x86_64 x86_64 GNU/Linux

Environment details (AWS, VirtualBox, physical, etc.):
physical

Steps to reproduce the error:
1. Install Ubuntu 15.10 and docker 1.10 on any machine
2. Run 1023 containers of ubuntu image
3. Now try to ping each container from the host

Actual Results: 
Not able to ping some containers from host.

Expected Results:
should ping every container from host.

Additional info:
we set the BR_PORT_BITS in br_private.h to 14 and recompiled the kernel.

**Comment by anthonyzxu** [comment]
I would check if the container is running first.

find container name for ip 172.17.0.6, 
cat /var/lib/docker/containers/*/hosts | grep -w "172.17.0.6"
172.17.0.6      7e1897226f7c  /*this is container name*/

check if it is running
docker ps | grep 7e1897226f7c
no output means it is not running

**Comment by mavenugo** [comment]
@rohit4813 I see that you are using a custom kernel with changes. I hope it is not interfering with the normal operation. But, assuming that is not the problem, can you please answer a few questions -

1. Can you find the container with the failing IP as @anthonyzxu suggested `cat /var/lib/docker/containers/*/hosts | grep -w "172.17.0.6" and actually checking the assigned ip inside the container namespace.

2. Do you trigger the ARP requests to multiple containers at the same time ? You might have to do some tuning on the ARP table thresholds : http://www.serveradminblog.com/2011/02/neighbour-table-overflow-sysctl-conf-tunning/

**Comment by rohit4813** [comment]
@mavenugo 
1. The container is running and is assigned the IP address 172.17.0.6.
2. No we have not triggered ARP requests to multiple containers at the same time.We were pinging only single container at a time.Moreover, there was no "Neighbour table overflow" error. 

When we run the command arp -an from the container bash having the ip 172.17.0.6 we get nothing.

**Comment by rohit4813** [comment]
So i tried running 1000 containers with the default linux kernel and still i am unable to ping some containers.

**Comment by aboch** [comment]
I think I was able to reproduce this. Initially all my 1024 containers are able to ping and can be pinged..
Only after minutes, the issues surfaced. It never recovered for the affected containers, they were not able to ping and host was not able to ping them.

Don't have more info ATM, need to debug further.

**Comment by neersighted** [comment]
Duplicate of https://github.com/moby/moby/issues/44973 as "many networks in one container" is effectively the same as "many containers in one network" from a kernel/netns/netdev perspective.
